android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonInstallationDialogFragment.kt

```diff
@@ -74,9 +74,9 @@ class AddonInstallationDialogFragment : AppCompatDialogFragment() {
     var onDismissed: (() -> Unit)? = null
 
     /**
-     * Reference to the application's [AddonCollectionProvider] to fetch add-on icons.
+     * Reference to the application's [DefaultAddonsProvider] to fetch add-on icons.
      */
-    var addonCollectionProvider: AddonCollectionProvider? = null
+    var addonsProvider: DefaultAddonsProvider? = null
```

Comment:
@Amejia481 your comment in the other patch (about abstraction and agnosticism) made me think about that: I don't think it is correct to have this property typed with the concrete implementation. IMO it should be the interface (`AddonsProvider`) but that is going to break some code because `getAddonIconBitmap()` isn't part of the interface (which is something else that I think should be fixed).

Which context am I missing here?

Reply:
Thanks for catching that, agree this is something that we missed in our initial implementation. It's not high priority we could file follow up bug and address it later as a refactor :)

Reply:
Thank you!